your_daily_dose_malware/main.py

```diff
@@ -1,12 +1,73 @@
+        json_response = response.json()
+        
+        if json_response['query_status'] == 'no_selector':
+            click.echo('	No selector provided. Please use either time or limit as selector',color=True)
+        if json_response['query_status'] == 'unknown_selector':
+            click.echo('	Selector unknown. Please use either time or limit as selector')
+        if json_response['query_status'] == 'no_results':
+            click.echo('	Your query yield no results')
+        else:
+            data_lenght = len(json_response['data'])
+            click.echo(f'	Your query yield {data_lenght} results')
+            for data in json_response['data']:
+                sha256_name = data['sha256_hash']
+                if os.path.exists(f'malware_{sha256_name[:4]}.zip'):
+                    click.echo(f'	{sha256_name} already downloaded no need to request again')
+                    continue
+                response = requests.post('https://mb-api.abuse.ch/api/v1/', data={'query':'get_file','sha256_hash':sha256_name}, headers=headers)
+                with open(f'malware_{sha256_name[:4]}.zip', 'wb+') as f:
+                    f.write(response.content)
+                    click.echo(f'	malware_{sha256_name[:4]}.zip downloaded')
+    elif by_sha256:
+        response  = requests.post('https://bazaar.abuse.ch/export/txt/sha256/recent', headers=headers)
+        with open('sha256_names.txt', 'wb+') as f:
+            f.write(response.content)
+            f.seek(0) # go back to the top of the file
+            new_hashes = list()
+            file_lines = [line.strip() for line in f.readlines()]
+            for index, line in enumerate(file_lines,start=0):
+                if index > 8 and index < len(file_lines)-1: # skip the first 9 lines and last line
+                    click.echo(line)
+                    new_hashes.append(line)
+                continue
+        with open('sha256_names.txt', 'w') as f:
+            for line in new_hashes:
+                f.write(line.decode('utf-8') + '\n')
+        sha256_names = open('sha256_names.txt', 'r').readlines()
+        click.echo(f'	{len(sha256_names)} hashes downloaded')
+        for sha256_hash in sha256_names:
+            if os.path.exists(f'malware_{sha256_hash[:4]}.zip'):
+                click.echo(f'	{sha256_hash} already downloaded no need to request again')
+                continue
+            response = requests.post('https://mb-api.abuse.ch/api/v1/', data={'query':'get_file','sha256_hash':sha256_hash}, headers=headers)
+            with open(f'malware_{sha256_hash[:4]}.zip', 'wb') as f:
+                f.write(response.content)
+            click.echo(f'	malware_{sha256_hash[:4]}.zip downloaded')
+    else:
+        click.echo('	No selector provided. Please use either by_sha256, hundred_recent as selector',)
 
 if __name__ == '__main__':
-    scraper.add_command(scraper)
+    scraper.add_command(run_scrapper)
```

Comment:
My prefered way to do CLIs, is not to use click, gut argsparse, collect args, and based on the args that the user enabled jump into some method or another from the beginning of the main script.

Reply:
Yes yes its could be a great idea too but that could be drastically change the structure cause we will no longer need commands packages

Reply:
But you don't  told this since the skeleton should we change this now cause if  yes that means change the structure of project and maybe we will no longer need commands  package

Reply:
We can keep it as is for now, but I have hard time seen how the cli tool will look like. 
We also need to standarize in a common approach to do this, I think argparse is more standard.